src/orion/core/worker/multi_task_algo.py

```diff
@@ -0,0 +1,551 @@
+        self.current_task_id = 0
+        assert "task_id" not in self._space
+        assert "task_id" not in self.space
+        # TODO: For some optimizers this exact key won't be found, there is a View(OneHotEncode(...))
+        # assert "task_id" in self.algorithm.transformed_space, self.algorithm.transformed_space
+        # assert "task_id" in self.algorithm.space
+        self.transformed_space = self.space
+
+    def suggest(self, num: int = 1) -> List[Point]:
+        """Suggest a `num` of new sets of parameters.
+
+        :param num: how many sets to be suggested.
+
+        .. note:: New parameters must be compliant with the problem's domain
+           `orion.algo.space.Space`.
+        """
+        points = self.algorithm.suggest(num=num)
+        if not self._enabled:
+            return points
+
+        # Remove the task ids from these points:
+        # NOTE: These predicted task ids aren't currently used.
+
+        points_without_task_ids = [self._remove_task_id(point) for point in points]
+        for point in points_without_task_ids:
+            if point not in self.space:
+                raise ValueError(
+                    textwrap.dedent(
+                        f"""\
+                        Point is not contained in space:
+                        Point: {point}
+                        Space: {self.space}
+                        """
+                    )
+                )
+        return points_without_task_ids
+
+    def observe(
+        self, points: List[Point], results: Union[List[float], List[Dict]]
+    ) -> None:
+        """Observe evaluation `results` corresponding to list of `points` in
+        space.
+
+        .. seealso:: `orion.algo.base.BaseAlgorithm.observe`
+        """
+        # Add the task ids to the points, because we assume that the points passed to
+        # `observe` are from the current task.
+        if self._enabled:
+            # Add the task ids to the points.
+            points = [
+                self._add_task_id(point, self.current_task_id) for point in points
+            ]
+        return self.algorithm.observe(points, results)
+
+    def warm_start(self, warm_start_trials: Dict[Mapping, List[Trial]]) -> None:
+        """ Use the given trials to warm-start the algorithm.
+
+        These experiments and their trials were fetched from some knowledge base, and
+        are believed to be somewhat similar to the current on-going experiment.
+
+        It is the responsability of the Algorithm to implement this method in order to
+        take advantage of these points.
+
+        Parameters
+        ----------
+        warm_start_trials : Dict[Mapping, List[Trial]]
+            Dictionary mapping from ExperimentInfo objects (dataclasses containing the
+            experiment config) to the list of Trials associated with that experiment.
+        """
+        # Being asked to warm-start the algorithm.
+        if is_warmstarteable(self.algorithm):
```

Comment:
It's a bit weird to have this function defined externally. I understand it is to support calling it with `str`, but maybe we could fix this once we get rid of the factor. A note for myself in a week or 2. :)